File: backends/java/zappy-adapters/src/main/java/com/zappymart/adapters/graphql/RevokeSessionPayload.java
package com.zappymart.adapters.graphql;

import com.zappymart.domain.accounts.Session;
import com.zappymart.domain.shared.UserError;

import java.util.List;

public record RevokeSessionPayload(List<Session> sessions, List<UserError> errors) {
}

File: backends/dotnet/src/Zappy.Adapters.GraphQL/Payloads/RevokeSessionPayload.cs
using Zappy.Domain;

namespace Zappy.Adapters.GraphQL;

public sealed class RevokeSessionPayload(IReadOnlyList<Session> sessions, IReadOnlyList<UserError> errors)
{
    public IReadOnlyList<Session> Sessions { get; } = sessions;

    public IReadOnlyList<UserError> Errors { get; } = errors;

    public static RevokeSessionPayload From(Zappy.Domain.Result<IReadOnlyList<Session>> result) =>
        new(result.Value ?? [], result.Errors);
}
